Fail fast

### Feature description

When required configuration are not satisfied, fail fast.

```
* What went wrong:
Execution failed for task ':app:analyzeProdReleaseBundle'.
> Error while evaluating property 'deviceSpec' of task ':app:analyzeProdnReleaseBundle'
   > Failed to calculate the value of task ':app:analyzeProdReleaseBundle' property 'deviceSpec'.
      > Cannot query the value of extension 'ruler' property 'abi' because it has no value available.
```

### Reasoning

```
BUILD FAILED in 7m 14s
```

--- a/ruler-gradle-plugin/src/main/kotlin/com/spotify/ruler/plugin/RulerPlugin.kt
+++ b/ruler-gradle-plugin/src/main/kotlin/com/spotify/ruler/plugin/RulerPlugin.kt
@@ -37,7 +37,7 @@ class RulerPlugin : Plugin<Project> {
                 val variantName = StringGroovyMethods.capitalize(variant.name)
                 project.tasks.register("analyze${variantName}Bundle", RulerTask::class.java) { task ->
                     task.appInfo.set(getAppInfo(project, variant))
-                    task.deviceSpec.set(getDeviceSpec(project, rulerExtension))
+                    task.deviceSpec.set(getDeviceSpec(rulerExtension))
 
                     task.bundleFile.set(variant.artifacts.get(SingleArtifact.BUNDLE))
                     task.mappingFile.set(variant.artifacts.get(SingleArtifact.OBFUSCATION_MAPPING_FILE))
@@ -56,16 +56,14 @@ class RulerPlugin : Plugin<Project> {
         AppInfo(
             applicationId = variant.applicationId.get(),
             versionName = variant.outputs.first().versionName.get() ?: "-",
-            variantName = variant.name
+            variantName = variant.name,
         )
     }
 
-    private fun getDeviceSpec(project: Project, extension: RulerExtension) = project.provider {
-        DeviceSpec(
-            abi = extension.abi.get(),
-            locale = extension.locale.get(),
-            screenDensity = extension.screenDensity.get(),
-            sdkVersion = extension.sdkVersion.get()
-        )
-    }
+    private fun getDeviceSpec(extension: RulerExtension) = DeviceSpec(
+        abi = extension.abi.orNull ?: error("ABI not specified."),
+        locale = extension.locale.orNull ?: error("Locale not specified."),
+        screenDensity = extension.screenDensity.orNull ?: error("Screen density not specified."),
+        sdkVersion = extension.sdkVersion.orNull ?: error("SDK version not specified."),
+    )
 }

--- a/ruler-gradle-plugin/src/test/kotlin/com/spotify/ruler/plugin/RulerIntegrationTest.kt
+++ b/ruler-gradle-plugin/src/test/kotlin/com/spotify/ruler/plugin/RulerIntegrationTest.kt
@@ -64,7 +64,8 @@ class RulerIntegrationTest {
         val buildGradle = projectDir.resolve("app/build.gradle")
         buildGradle.writeText(buildGradle.readText().substringBefore("ruler {"))
 
-        gradlew(task, expectFailure = true)
+        val tasks = gradlew(task, expectFailure = true).tasks
+        assertThat(tasks).isEmpty() // We want to fail early
     }
 
     @ParameterizedTest

